### ISSUE TITLE
feature: add ignore directory option to cli

### ISSUE THREAD

**Issue by Marvin9** [vale-cli/vale#243]
One of spotify's project [backstage](https://github.com/spotify/backstage) recently adopted vale and It's impressive! Thank you for that. However if you visit this [pull request](https://github.com/spotify/backstage/pull/2631), I had to write cross platform verbose script just to ignore ```node_modules```. It would be great if there is an option to ignore directories or ignore few pre known directories by default eg. node_modules, .git.

**Comment by jdkato** [comment]
If your goal is to run Vale on Markdown files that are checked into Git, you can just use `git ls-files | xargs vale`. This will give equivalent results to your current `yarn lint:docs` command on both Unix and Windows (*Git for Windows* includes `xargs`) without the need for the "verbose" Node.js wrapper script.

---

Regarding the feature request itself, my general thoughts are:

1. Supporting an existing VCS ignore file (such as `.gitignore`) is beyond the scope of Vale. I don't think it's worth the development / maintenance cost when the alternative is as simple as something like `git ls-files | xargs vale`.

2. Adding something like `--ignore=<folder1>,<folder2>` feels like sort of a half-baked solution to a much more general problem. It's inevitably going to lead to needing to ignore arbitrary patterns.

3. This could be done by extending the functionality of the existing `--glob` flag, which is currently based on [`gobwas/glob`](https://github.com/gobwas/glob) and operates on files rather than folders. This is something that I'd like to do (even if only to make `--glob` more standards-compliant), but there aren't any *complete* libraries available for doing this in Go at the moment.

So, in short, you'll eventually be able to ignore directories through the `--glob` flag, but I'm not sure when I'll have time to implement it.

**Comment by Marvin9** [comment]
Thanks. 3rd point would be really helpful. One more thing I would like to point out is there any chance to ignore directories out of the box? As mentioned in issue eg node_modules, .git. 

Can you also please point out where is vale ```walking``` the directories, the same way it ```walks``` files? I want to try it out.

**Comment by jdkato** [comment]
It's right here:

https://github.com/errata-ai/vale/blob/b3f7570ee19178479aaac44f9c114ec4ced386e4/lint/lint.go#L124-L129

We could do something like

```go
if f.IsDir() && f.Name() == ".git" {
    return filepath.SkipDir
}
```

as a temporary solution.

**Comment by Marvin9** [comment]
Thanks. This(mentioned below) was my approach on similar issue. I will be making PR soon.

https://github.com/Marvin9/licensor/blob/master/utils/constants.go#L63

**Comment by Marvin9** [comment]
Thanks @jdkato
